client/src/components/DataFiles/DataFilesModals/DataFilesLinkModal.js

```diff
@@ -0,0 +1,220 @@
+import React, { useState, useCallback } from 'react';
+import { useSelector, useDispatch } from 'react-redux';
+import {
+  Button,
+  Modal,
+  ModalHeader,
+  ModalBody,
+  ModalFooter,
+  Form,
+  FormGroup,
+  Label
+} from 'reactstrap';
+import PropTypes from 'prop-types';
+import { LoadingSpinner, Message, TextCopyField, InlineMessage } from '_common';
+import './DataFilesLinkModal.module.scss';
+import './DataFilesLinkModal.scss';
+
+const statusPropType = PropTypes.shape({
+  error: PropTypes.string,
+  url: PropTypes.string,
+  method: PropTypes.string,
+  loading: PropTypes.bool
+});
+
+const DataFilesLinkActions = ({ status, onClick }) => {
+  const disabled = status && status.method != null;
+
+  if (status && status.url) {
+    return (
+      <>
+        <Button
+          disabled={disabled}
+          className="data-files-btn"
+          onClick={e => onClick('delete')}
+          styleName="action"
+        >
+          Delete
+        </Button>
+        <Button
+          disabled={disabled}
+          className="data-files-btn"
+          onClick={e => onClick('put')}
+        >
+          Replace Link
+        </Button>
+      </>
+    );
+  }
+
+  return (
+    <Button
+      disabled={disabled}
+      className="data-files-btn"
+      onClick={e => onClick('post')}
+    >
+      Generate Link
+    </Button>
+  );
+};
+
+DataFilesLinkActions.propTypes = {
+  status: statusPropType,
+  onClick: PropTypes.func.isRequired
+};
+
+DataFilesLinkActions.defaultProps = {
+  status: {
+    error: null,
+    url: '',
+    method: null,
+    loading: false
+  }
+};
+
+const DataFilesLinkStatus = ({ status }) => {
+  if (!status) {
+    return null;
+  }
+  if (status.loading) {
+    return <LoadingSpinner placement="inline" />;
+  }
+  if (status.error) {
+    // Error occurred during retrieval of link
+    return (
+      <Message type="error">
+        There was a problem retrieving the link for this file.
+      </Message>
+    );
+  }
+  return (
+    <FormGroup>
+      <Label>Link</Label>
+      <TextCopyField
+        placeholder="Click generate to make a link"
```

Comment:
```suggestion
        placeholder='Click "Generate Link" to make a link'
```

Reply:
The placeholder text is to be removed, and instead be help text beneath the field.

FP-740 image: [Late Design - Modal - New.png - Latest 12/11/2020 3:34 PM - Wesley Bomar](https://jira.tacc.utexas.edu/secure/attachment/14721/Late%20Design%20-%20Modal%20-%20New.png)

Reply:
@tacc-wbomar short term do we remove the placeholder then?

Reply:
Not short term. Forever.

I reword my message:

Please, remove the placeholder text. It should not be present.

Reference:

- [Late Design - Modal - New.png - Latest 12/11/2020 3:34 PM - Wesley Bomar](https://jira.tacc.utexas.edu/secure/attachment/14721/Late%20Design%20-%20Modal%20-%20New.png) _(**Notice**: The help text in this image **is not** accurate. The help text in the code "This link downloads the file without requiring login or an account." **is** accurate.)_